### ISSUE TITLE
With remote option table is still sorting incoming data

### ISSUE THREAD

**Issue by jozarz** [react-bootstrap-table/react-bootstrap-table2#335]
```
import React, { Component } from "react";
import BootstrapTable from "react-bootstrap-table-next";

const columns = [
  { dataField: "id", text: "id", sort: true },
  { dataField: "name", text: "Full name" },
  { dataField: "email", text: "Email" },
  { dataField: "phone", text: "Phone" },
  { dataField: "website", text: "Website" }
];

const data = [
  {
    id: 1,
    name: "Leanne Graham",
    username: "Bret",
    email: "Sincere@april.biz",
    phone: "1-770-736-8031 x56442",
    website: "hildegard.org",
  },
  {
    id: 2,
    name: "Ervin Howell",
    username: "Antonette",
    email: "Shanna@melissa.tv",
    phone: "010-692-6593 x09125",
    website: "anastasia.net",
  },
  {
    id: 3,
    name: "Clementine Bauch",
    username: "Samantha",
    email: "Nathan@yesenia.net",
    phone: "1-463-123-4447",
    website: "ramiro.info",
  },
  {
    id: 4,
    name: "Patricia Lebsack",
    username: "Karianne",
    email: "Julianne.OConner@kory.org",
    phone: "493-170-9623 x156",
    website: "kale.biz",
  },
  {
    id: 5,
    name: "Chelsey Dietrich",
    username: "Kamren",
    email: "Lucio_Hettinger@annie.ca",
    phone: "(254)954-1289",
    website: "demarco.info",
  },
  {
    id: 6,
    name: "Mrs. Dennis Schulist",
    username: "Leopoldo_Corkery",
    email: "Karley_Dach@jasper.info",
    phone: "1-477-935-8478 x6430",
    website: "ola.org",
  },
  {
    id: 7,
    name: "Kurtis Weissnat",
    username: "Elwyn.Skiles",
    email: "Telly.Hoeger@billy.biz",
    phone: "210.067.6132",
    website: "elvis.io",
  },
  {
    id: 8,
    name: "Nicholas Runolfsdottir V",
    username: "Maxime_Nienow",
    email: "Sherwood@rosamond.me",
    phone: "586.493.6943 x140",
    website: "jacynthe.com",
  },
  {
    id: 9,
    name: "Glenna Reichert",
    username: "Delphine",
    email: "Chaim_McDermott@dana.io",
    phone: "(775)976-6794 x41206",
    website: "conrad.com",
  },
  {
    id: 10,
    name: "Clementina DuBuque",
    username: "Moriah.Stanton",
    email: "Rey.Padberg@karina.biz",
    phone: "024-648-3804",
    website: "ambrose.net",
  }
];

class Wrapper extends Component {
  state = { data: data }

  onTableChange = () => {
    this.setState({ data: data })
  }

  render() {
   return(
     <div style={{ display: "flex", justifyContent: "center" }}>
       <BootstrapTable
         columns={columns}
         data={this.state.data}
         totalSize={data.length}
         remote={{sort: true, pagination: true }}
         keyField="id"
         onTableChange={this.onTableChange}
       />
     </div>
   )
  }
}
```
It is just small example. After invoking sort i except that incoming data is not rearranged by component. This example might be odd but we had some problems with sorting data by date column (on backend but it is displayed as string in table) and somehow rows were out of order.

possible fix:
https://github.com/react-bootstrap-table/react-bootstrap-table2/blob/master/packages/react-bootstrap-table2/src/sort/wrapper.js#L41

`componentWillReceiveProps` should do nothing if `this.isRemoteSort() || this.isRemotePagination()`

**Comment by AllenFang** [comment]
@jozarz thanks, it probably should not do sorting. I will do some enhancement.

**Comment by amoludage** [comment]
@AllenFang any update on this issue ?

Can I add PR to fix this?

**Comment by AllenFang** [comment]
@amoludage sure, any PR is welcome ;)

**Comment by AllenFang** [comment]
https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2018/06/04/version-bump.html

Fixed on `react-bootstrap-table-next@0.1.13`
